lib/primitives/element.js

```diff
@@ -94,18 +95,19 @@ var Element = createClass({
   },
 
   /// Finds the given elements in the element tree
-  /// Returns ArrayElement of elements
+  /// Returns ElementSlice of elements
   /// findRecursive: function(names..., parents)
   findRecursive: function() {
     var ArrayElement = require('./array-element');
     var elementNames = [].concat.apply([], arguments);
-    var array = new ArrayElement();
+    var elements = new ElementSlice();
     var parents;
 
-    if (elementNames[elementNames.length - 1].element === 'array') {
-      parents = elementNames.pop().clone();
+    if (elementNames[elementNames.length - 1] instanceof ElementSlice) {
+      // clone the given parents
+      parents = new ElementSlice(elementNames.pop().map(function (e) { return e; }));
```

Comment:
This looks a little messy, but it is exactly how `clone()` on element was working for an ArrayElement, by using `map` to create a copy. I have kept same behaviour for consistency.

This line and surrounding code will be removed in a future PR I am working on to add parent tracking to elements so that we don't need to track parents and make clones here.